Feature/salt pepper

vSebas opened the pull request.

vSebas committed:
Added salt and pepper feature. This commit only supports white dots

vSebas committed:
Salt and pepper feature with black and white dots

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -21,6 +21,8 @@ int main( int argc, char** argv )
    //Apply average filter
    data.AverageFilter(kernel_size);
+   //Add salt and pepper noise
+   data.SaltPepper(0.2);
```
el parametro de %  definelo como variale al inicio del programa, nada de "Magic Numbers"

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -37,4 +37,29 @@ void DataAugmentation::AverageFilter(const int &kernel){
+}
+
+void DataAugmentation::SaltPepper(const float percentage){
+  cv::Mat pepperized_ = in_.clone();
```
En lugar de crear una nueva matriz llamada pepperized_, usa la matriz ya establecida out_

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -37,4 +37,29 @@ void DataAugmentation::AverageFilter(const int &kernel){
+  cv::Mat pepperized_ = in_.clone();
+  int total = percentage * pepperized_.cols * pepperized_.rows;
+  for(int i = 0; i<total; i++){
+    //Choose a random pixel between 0 - (im.cols-1)
```
modifica comentario a que indique el nombre de la matriz que estas usando. 
Exemplo: between 0 and (out_.cols-1)

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -37,4 +37,29 @@ void DataAugmentation::AverageFilter(const int &kernel){
+  for(int i = 0; i<total; i++){
+    //Choose a random pixel between 0 - (im.cols-1)
+    //and 0 - (im.rows-1)
+    int a =rand()%pepperized_.cols;
```
Cambia el nombre de la varibles a y b a algo mas descriptivo. 
Ejemplo: row and column 
row_pixel and column_pixel

vSebas replied:
Done

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -21,6 +21,8 @@ int main( int argc, char** argv )
    data.Read("../../imgs/lena.png");
    //Apply average filter
    data.AverageFilter(kernel_size);
+   //Add salt and pepper noise
```
Modifica los headers the todos los files para incluir tu nombre y la fecha en la que se modifico el file

vSebas replied:
Considero que es mejor hacerlo una vez se haga merge a todos los pull request

ivanacollg commented:
El branch tiene conflictos con el master, corre desde  tu terminal y desde este branch `git pull origin master`. Espero que asi se resuelvan los conflictos, si no, vas a tener que hacer un rebase o un merge

vSebas committed:
Made these changes:
noise_percentage was passed as parameter in unit test
fixed comments
changed the name of some variables
used out_ matrix instead of pepperized_ matrix

ivanacollg commented:
Arregla merge conflicts

vSebas committed:
Updated brief and added parameter comment

vSebas committed:
Added Ivana's mail

vSebas committed:
Merge branch 'master' into feature/salt_pepper

ivanacollg committed:
Merge branch 'master' of github.com:vanttec/data_augmentation into feature/salt_pepper
